packages/eslint-plugin/src/rules/camelcase.ts

```diff
@@ -109,8 +145,11 @@ export default util.createRule<Options, MessageIds>({
 
         // Check TypeScript specific nodes
         const parent = node.parent;
-        if (parent && isTSPropertyType(parent)) {
-          if (properties === 'always' && isUnderscored(name)) {
+        if (parent && (isTSPropertyType(parent) || isTSTypeParameter(parent))) {
+          if (
+            [properties, genericType].includes('always') &&
+            isUnderscored(name)
+          ) {
```

Comment:
won't this allow you to inadvertently mix and match settings?

I.e. if I set `properties: "always"`, and then lint `function foo<t>() {}`, won't I get a lint failure?

You probably need to split it out into two separate checks